src/deepsparse/pipeline.py

```diff
@@ -194,6 +194,40 @@ def __call__(self, *args, **kwargs) -> Union[BaseModel, Future]:
             else self._run(*args, **kwargs)  # Blocking call
         )
 
+    @staticmethod
+    def get_task_constructor(task: str) -> Type["Pipeline"]:
+        """
+        This function retrieves the class previously registered via `Pipeline.register`
+        for `task`.
+
+        If `task` starts with "custom", then it is mapped to the "custom" task.
+
+        :param task: The task name to get the constructor for
+        :return: The class registered to `task`
+        :raises ValueError: if `task` was not registered via `Pipeline.register`.
+        """
+        task = task.lower().replace("-", "_")
+
+        # support any task that has "custom" at the beginning via the "custom" task
+        if task.startswith("custom"):
+            task = "custom"
```

Comment:
This new function is pulled from Pipeline.create() (which now uses this function). except for these two lines, which allow support for any task starting with "custom".

Pulling this function out made the unit tests  a bit faster since they don't have to go through compiling the model via Pipeline.create

Reply:
let's avoid breaking this funcitonality out into a new API, we still want users going through `.create`. additionally, for mapping `custom` tasks.

additionally, what's the motivation for making 'custom' a catch-all. if needed, can we add logic for that in SupportedTask / task.py instead? want to avoid too much casework in the base pipeline file (breaking out the bloat in this file is towards the top of our tech debt)

Reply:
I can make it a private function - this one is mainly intended to make unit testing easier.

The issue with supported task was that Pipeline.create does the check for `task not in _REGISTERED_PIPELINES`. SupportedTask can pass anything that starts with custom, but it'd have to register that task somehow into that dictionary.